### PR TITLE
Partial ideal step refactor and updated error messages

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR* [version update](https://semver.org/)

(Minor version due to the [substantial improvements within the private code](https://semver.org/#spec-item-7))

### Context

This PR updates the Step's error messages, to achieve it easily, the Step got a partial ideal Step refactor too.

### Changes

- Top-level input parsing errors got a new format: `Process config: %s`
- Top-level step dependency installation warning got a new format: `Install dependencies: %s`
- Top-level step run errors got a new format: `Run: %s`
- Top-level step dependency install got a new format: `Export outputs: %s`
- Improved error messages
  - removed `, error: ` suffixes from error messages
  - use `%w` for wrapping errors

Non-functional changes:
- Partial Ideal Step refactor:
  - main function is split into: `TestBuilder.ProcessConfig`, `TestBuilder.InstallDependencies`, `TestBuilder.Run` and `TestBuilder.ExportOutput`
  - the step's functionality is moved to step.go, main.go contains only glue code

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	log.Infof(colorstring.Magenta(`
+The log file is stored in $BITRISE_DEPLOY_DIR, and its full path
+is available in the $BITRISE_XCODEBUILD_TEST_LOG_PATH environment variable.
+If you have the Deploy to Bitrise.io step (after this step),
+that will attach the file to your build as an artifact!`))
+}
+
+func exportXcodebuildTestLog(deployDir, xcodebuildTestLog string) error {
+	pth, err := saveRawOutputToLogFile(xcodebuildTestLog)
```
Comment (lpusok) [inline utils.go:38]: Why save the file to a temporary path first, instead of writing it in the final location?

File: utils.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		log.Infof(lastLines)
+	}
+
+	fmt.Println(stringutil.LastNLines(rawXcodebuildOutput, 20))
+
+	if !isRunSuccess {
+		log.Warnf("If you can't find the reason of the error in the log, please check the xcodebuild_test.log.")
+	}
+
+	log.Infof(colorstring.Magenta(`
```
Comment (lpusok) [inline utils.go:30]: The message does not match with the name of the actual constant (BITRISE_XCODE_RAW_RESULT_TEXT_PATH); I recall there was a dsicussion in the earlier Inptus revamp to not cahnge that, but do not remember the exact reasons.

File: step.go [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+type ExportOpts struct {
+	RunOut
+	OutputDir string
+}
+
+func (b TestBuilder) ExportOutput(opts ExportOpts) error {
+	log.Infof("Export:")
+
+	if opts.XcodebuildLog != "" {
+		if err := exportXcodebuildTestLog(opts.OutputDir, opts.XcodebuildLog); err != nil {
```
Comment (lpusok) [inline step.go:308]: Is it worth printing the location of the raw build log? In some cases printLastLinesOfXcodebuildTestLog mentions the env var, but not in all cases.

**Review by lpusok — CHANGES_REQUESTED** [review]

File: utils.go (hunk 2 above)
Comment (godrei) [reply utils.go]: Kept the original env var (`BITRISE_XCODE_RAW_RESULT_TEXT_PATH`) and updated these logs to print the truth.

File: step.go (hunk 3 above)
Comment (godrei) [reply step.go]: For the sake of simplicity, I would print all the exported outputs in this phase.

File: step.go [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+		Dir: opts.SYMRoot,
+	})
+	if out, err := zipCmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
+		var exerr *exec.ExitError
+		if errors.As(err, &exerr) {
+			return fmt.Errorf("%s failed: %s", zipCmd.PrintableCommandArgs(), out)
+		}
+		return fmt.Errorf("%s failed: %w", zipCmd.PrintableCommandArgs(), err)
+	}
+	log.Printf("Zipped test bundle: %s", outputTestBundleZipPath)
```
Comment (lpusok) [inline step.go:346]: This log is not needed as the same info is printed below.

File: step.go [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+	defer func() {
+		if authOptions != nil && authOptions.KeyPath != "" {
+			if err := os.Remove(authOptions.KeyPath); err != nil {
+				log.Warnf("failed to remove private key file: %s", err)
+			}
+		}
+	}()
+
+	// Build for testing
+	log.Infof("Build:")
```
Comment (lpusok) [inline step.go:249]: NOTE: add a fmt.Println() before, for better seperated log output. (Same for Export phase.)

File: step.go [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+		if swiftPackagesPath, err = cache.SwiftPackagesPath(absProjectPath); err != nil {
+			return Config{}, fmt.Errorf("failed to get swift packages path: %w", err)
+		}
+	}
+
+	var customOptions []string
+	if input.XcodebuildOptions != "" {
+		customOptions, err = shellquote.Split(input.XcodebuildOptions)
+		if err != nil {
+			return Config{}, fmt.Errorf("failed to parse additional options (%s): %w", input.XcodebuildOptions, err)
```
Comment (lpusok) [inline step.go:143]: Used something like "provided additional options are not valid CLI argumensts" in the npm/yarn Steps on Oliver's reccomendation.

**Review by lpusok — CHANGES_REQUESTED** [review]

**Review by lpusok — APPROVED** [review]

**Pull request merged**